src/commands/deployImage/getRegistryCredentialsAndSecrets.ts

```diff
@@ -3,24 +3,24 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import type { ContainerApp, RegistryCredentials, Secret } from "@azure/arm-appcontainers";
+import type { RegistryCredentials, Secret } from "@azure/arm-appcontainers";
 import { nonNullProp } from "@microsoft/vscode-azext-utils";
 import { dockerHubDomain, dockerHubRegistry } from "../../constants";
-import { listCredentialsFromRegistry } from "./acr/listCredentialsFromRegistry";
 import { IDeployImageContext } from "./IDeployImageContext";
+import { listCredentialsFromRegistry } from "./acr/listCredentialsFromRegistry";
 
 interface RegistryCredentialsAndSecrets {
-    registries: RegistryCredentials[] | undefined;
-    secrets: Secret[] | undefined;
+    registries?: RegistryCredentials[];
+    secrets?: Secret[];
 }
 
-export async function getAcrCredentialsAndSecrets(context: IDeployImageContext, containerAppEnvelope: Required<ContainerApp>): Promise<RegistryCredentialsAndSecrets> {
+export async function getAcrCredentialsAndSecrets(context: IDeployImageContext, containerAppSettings?: RegistryCredentialsAndSecrets): Promise<RegistryCredentialsAndSecrets> {
     const registry = nonNullProp(context, 'registry');
     const { username, password } = await listCredentialsFromRegistry(context, registry);
     const passwordName = `${registry.name?.toLocaleLowerCase()}-${password?.name}`;
 
     // Remove duplicate registries
-    const registries: RegistryCredentials[] | undefined = containerAppEnvelope.configuration.registries?.filter(r => r.server !== registry.loginServer);
+    const registries: RegistryCredentials[] = containerAppSettings?.registries?.filter(r => r.server !== registry.loginServer) ?? [];
```

Comment:
Was this a necessary change? I don't know if `undefined` and `[]` behave the exact same in the context of a container app site envelope.

Reply:
I bet we could go either way.  It appears to me that the two are probably interchangeable, but the reason I added that is because there is a section in `ContainerAppCreateStep` that looks like this:

<img width="215" alt="image" src="https://user-images.githubusercontent.com/40250218/223573567-244bb225-1d36-4bee-b804-871f35263d15.png">

This would preserve that assignment behavior